### PR TITLE
Fix/high cold start times

### PR DESCRIPTION
**Pull request by portdeveloper**
### This PR

- Removes unused code
- Uses a custom search bar
- Removes the use of catch-all route
- Adds a skeleton to total balances and address card
- Removes NextNProgress
- Improves overall code quality
- Improves styling

## TO-DO (may tackle these in a separate PR

- [ ] add a tooltip to copy icon instead of changing to checkmark
- [ ] extract ens-getting logic into a custom hook
- [ ] totalbalances card skeletons could have lighter colors
- [ ] blockscan link does not work, may change it to basescan?

If vercel fails to deploy, you can see the branch live here: https://test-vision.vercel.app

## Additional Information

- [ ] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [ ] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #63 

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

### CODE REVIEW

File: packages/nextjs/components/address-vision/AddressCard.tsx [hunk 1]
```diff
@@ -1,30 +1,123 @@
...
 
+  const blockExplorerLink = getBlockExplorerAddressLink(address);
+  let displayAddress = address?.slice(0, 5) + "..." + address?.slice(-4);
+
+  if (ens) {
+    displayAddress = ens;
+  }
+
+  const size = getSize(displayAddress);
+  const textSizeClass = `text-${size}`;
```
Comment (rin-st) [inline packages/nextjs/components/address-vision/AddressCard.tsx:83]: may not work as expected https://tailwindcss.com/docs/content-configuration#dynamic-class-names

**Review by Pabl0cks — COMMENTED** [review]
Something weird happens when you change from one address with ENS associated to an address with no ENS .

If you change from no-ens to no-ens works fine, and from ens to ens works fine too (no matter if you paste the 0x or the .eth)

Example where I'm just pasting an address with no ens, having vitalik.eth as previous search (`0x6b3a8798e5fb9fc5603f3ab5ea2e8136694e55d0` for example):

Change from  eth to another 0x address

File: packages/nextjs/components/address-vision/AddressCard.tsx (hunk 1 above)
Comment (rin-st) [reply packages/nextjs/components/address-vision/AddressCard.tsx:83]: I think returning full class name like `text-3xl` from `getSize` will fix it

Comment (portdeveloper) [reply packages/nextjs/components/address-vision/AddressCard.tsx:83]: It is a bad practice but it seems to work

**Comment by Pabl0cks** [comment]
The swap between ENS and no-ENS search is working great now! TYSM 🙌

I've found another issue with ENS addresses.

After you've searched an ENS address, if you try to delete the current search, using return key, it doesn't work, even spamming. The only way is to keep "return" pressed. 

It doesn't work even if you select all the ENS and hit "del" or "return". You have to select all ENS and paste another address if you want to make this other method work.

A couple videos:

<details><summary>Trying to delete spamming return</summary>

ens-update-delete
</details>

<details><summary>Select all and hit del or return</summary>

select-all-delete
</details>

**Comment by Pabl0cks** [comment]
When you search for an address, the search bar stays in edition mode, which in mobile causes keyboard to stay opened after the search.

image

On previous version you'd loose the focus of the search field and this keyboard issue did not happen.

**Comment by portdeveloper** [comment]
Thanks for the help @Pabl0cks @rin-st !!!

Let me merge this and I will fix any further bugs in prod.

**Pull request merged**